packages/realm/bindgen/src/templates/base-wrapper.ts

```diff
@@ -0,0 +1,162 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+import { strict as assert } from "assert";
+import { Property } from "@realm/bindgen/bound-model";
+import { TemplateContext } from "@realm/bindgen/context";
+import { Outputter } from "@realm/bindgen/outputter";
+
+import { doJsPasses } from "../js-passes";
+
+export function generateNativeBigIntSupport(out: Outputter) {
```

Comment:
I refactored the "node-wrapper" into two separate templates, both with their shared code in this "base-wrapper".